docs(tutorial): Plausible Web Analytics on Ubuntu (#1338)
scaleway/docs-content: tutorials/plausible-analytics-ubuntu/index.mdx

---
meta:
  title: Running web analytics with Plausible on Ubuntu Linux
  description: This page shows how to generate web analytics with Plausible on Cost-Optimzed Instances running Ubuntu Linux
content:
  h1: Running web analytics with Plausible on Ubuntu Linux
  paragraph: This page shows how to generate web analytics with Plausible on Cost-Optimzed Instances running Ubuntu Linux
tags: compute instance plausible ubuntu analytics open-source
hero:
categories:
  - compute
  - instance
dates:
  validation: 2023-02-21
  posted: 2023-02-21
---

> Plausible Analytics is an open-source web analytics project with the mission to make analytics more privacy-friendly. You can self-host the solution on Scaleway Instances and to reduce the dependency and surveillance of other tools coming from the AdTech world.

Reviewer: ```suggestion
Plausible Analytics is an open-source web analytics project with the mission to make analytics more privacy-friendly. You can self-host the solution on Scaleway Instances and reduce the dependency and surveillance of other tools coming from the AdTech world.
```

    ```
5. Update the packet manager, and install Docker and its dependencies using `apt`.
    ```
    apt update && apt install docker-ce docker-ce-cli docker-compose containerd.io docker-buildx-plugin docker-compose-plugin
    ```
6. Download and unpack the Plausible repository:
    ```
    curl -L https://github.com/plausible/hosting/archive/master.tar.gz | tar -xz
    ```
    <Message type="tip">
      If you have `git` installed on your machine, you can also clone the repository using the following command:
      ```
      git clone https://github.com/plausible/hosting
      ```
    </Message>
7. Enter the Plausible directory:
    ```
    cd hosting*
    ```
    Two important files are located in the downloaded folder:
    `docker-compose.yml` - installs and orchestrates networking between your Plausible server, Postgres database, Clickhouse database (for stats), and an SMTP server. This file comes preconfigured with values for most setups. You can tweak it if required.
    `plausible-conf.env` - configures the Plausible server itself. The full list of configuration options is available in the [official documentation](https://plausible.io/docs/self-hosting-configuration).

> 8. Generate a random 64-character secret key which will be used to secure the application. Run the following command to generate one and copy it into your computers clipboard.

Reviewer: ```suggestion
8. Generate a random 64-characters secret key which will be used to secure the application. Run the following command to generate one and copy it into your computer's clipboard.
```

    ```
    openssl rand -base64 64 | tr -d '\n' ; echo
    ```
9. Open the file in a text editor (e.g. `nano`) and set the `SECRET_KEY_BASE` to the key you have generated in the previous step. Then set the `BASE_URL` for your Instance. Optionally include a port, if no port is configured it will use the default port `8000`.
    ```
    nano plausible-conf.env
    ```
10. Start the server
    ```
    docker-compose up -d
    ```

## Configuring a Nginx reverse and caching proxy for Plausible

> As Plausible supports by default plain HTTP traffic by only, we configure a Nginx reverse proxy to provide HTTPS with a Let's Encrypt certificate to increase security and cache the analytics script for improved performance.

Reviewer 2: ```suggestion
As Plausible supports only plain HTTP traffic by default, we configure a Nginx reverse proxy to provide HTTPS with a Let's Encrypt certificate to increase security and cache the analytics script for improved performance.
```

      ```
    </Message>
7. Enter the Plausible directory:
    ```
    cd hosting*
    ```
    Two important files are located in the downloaded folder:
    `docker-compose.yml` - installs and orchestrates networking between your Plausible server, Postgres database, Clickhouse database (for stats), and an SMTP server. This file comes preconfigured with values for most setups. You can tweak it if required.
    `plausible-conf.env` - configures the Plausible server itself. The full list of configuration options is available in the [official documentation](https://plausible.io/docs/self-hosting-configuration).
8. Generate a random 64-characters secret key which will be used to secure the application. Run the following command to generate one and copy it into your computer's clipboard.
    ```
    openssl rand -base64 64 | tr -d '\n' ; echo
    ```
9. Open the file in a text editor (e.g. `nano`) and set the `SECRET_KEY_BASE` to the key you have generated in the previous step. Then set the `BASE_URL` for your Instance. Optionally include a port, if no port is configured it will use the default port `8000`.
    ```
    nano plausible-conf.env
    ```

> 10. Start the server

Reviewer 3: ```suggestion
10. Start the server.
```

    ```
    docker-compose up -d
    ```

> ## Configuring a Nginx reverse and caching proxy for Plausible

Reviewer 4: ```suggestion
## Configuring an Nginx reverse and caching proxy for Plausible
```

> As Plausible supports only plain HTTP traffic by default, we configure a Nginx reverse proxy to provide HTTPS with a Let's Encrypt certificate to increase security and cache the analytics script for improved performance.

Reviewer 4: ```suggestion
As Plausible supports only plain HTTP traffic by default, we will configure an Nginx reverse proxy to provide HTTPS with a Let's Encrypt certificate to increase security and cache the analytics script for improved performance.
```
